libRALFit/src/ral_nlls_internal.f90

```diff
@@ -3245,7 +3552,8 @@ subroutine solve_spd(A,b,LtL,x,n,inform)
        inform%status = 0
        ! wrapper for the lapack subroutine dposv
        LtL(1:n,1:n) = A(1:n,1:n)
-       x(1:n) = b(1:n)
+       ! NOTE: b is inverted!
```

Comment:
b is actually -b so it solves Ax=-b, this also applies to solve_general and solve_spd_nocopy (STP also extended), this avoid auto allocs in all the places that calls solve_xxx(..., A, -v, ...)

Reply:
@tyronerees suggested a name change. e.g. in `solve_spd` to account for the inverting of `b` in `Ax=-b`.